python/src/deltachat/direct_imap.py

```diff
@@ -25,14 +25,14 @@ def __init__(self, account):
         self.select_folder("INBOX")
 
     def shutdown(self):
-        try:
-            self.connection.close()
-        except Exception:
-            pass
         try:
             self.connection.logout()
         except Exception:
             print("Could not logout direct_imap conn")
+        try:
+            self.connection.close()
+        except Exception:
+            pass
```

Comment:
From https://docs.python.org/3/library/imaplib.html: 
```
IMAP4.close()

    Close currently selected mailbox. Deleted messages are removed from writable mailbox. This is the recommended command before LOGOUT.
```
Call close (will actually unselect the selected folder IIRC), then logout.

Reply:
fixed